addon/session-stores/cookie.js

```diff
@@ -198,5 +257,14 @@ export default BaseStore.extend({
     } else {
       return RSVP.resolve();
     }
+  },
+
+  // FIXME: maybe be causing getter bugs with rewriteCookie called before
+  // setter can return value
```

Comment:
I see what you mean - good point. We should maybe wrap `rewriteCookie` in `Ember.run.scheduleOnce` anyway so we don't rewrite unnecessarily often which should also solve this problem.